packages/atlas/src/components/_crt/CrtRevenueShareWidget/CrtRevenueShareWidget.tsx

```diff
@@ -0,0 +1,160 @@
+  token: FullCreatorTokenFragment
+  onTabSwitch?: () => void
+}
+
+export const CrtRevenueShareWidget = ({ token, onTabSwitch }: CrtHoldersWidgetProps) => {
+  const activeRevenueShare = token.revenueShares.find((rS) => !rS.finalized)
+  const { convertBlockToMsTimestamp } = useBlockTimeEstimation()
+  const endingBlockTimestamp = convertBlockToMsTimestamp(activeRevenueShare?.endsAt ?? 0)
+  const endingDate = endingBlockTimestamp ? new Date(endingBlockTimestamp) : null
+  const status: 'active' | 'past' | 'inactive' =
+    !endingBlockTimestamp || !activeRevenueShare ? 'inactive' : endingBlockTimestamp < Date.now() ? 'past' : 'active'
+
+  return (
+    <Widget
+      title="Revenue share with holders"
+      titleVariant="h500"
+      titleColor="colorTextStrong"
+      customTopRightNode={
+        <TextButton iconPlacement="right" onClick={onTabSwitch} icon={<SvgActionChevronR />}>
+          Show revenue shares
+        </TextButton>
+      }
+      customNode={
+        <FlexBox flow="column" justifyContent="space-between" height="100%" width="100%" gap={4}>
+          <FlexBox width="100%" equalChildren justifyContent="space-between">
+            <FlexBox flow="column">
+              <Text variant="h100" as="h1" color="colorTextMuted">
+                {status === 'inactive'
+                  ? 'CURRENT STATE'
+                  : status === 'past'
+                  ? 'REVENUE SHARE ENDED ON'
+                  : 'REVENUE SHARE ENDS IN'}
+              </Text>
+              {status !== 'inactive' && activeRevenueShare?.endsAt ? (
+                status === 'past' ? (
+                  <Text variant="h400" as="h5" margin={{ bottom: 4 }}>
+                    {endingDate ? formatDateTime(endingDate).replace(',', ' at') : 'N/A'}
+                  </Text>
+                ) : (
+                  <Text variant="h400" as="h5">
+                    {endingDate ? formatDurationShort(Math.round((endingDate.getTime() - Date.now()) / 1000)) : 'N/A'}
+                  </Text>
+                )
+              ) : (
+                <Text variant="h400" as="h5" margin={{ bottom: 4 }}>
+                  No active share
+                </Text>
+              )}
+            </FlexBox>
+            <FlexBox flow="column">
+              <Text variant="h100" as="h1" color="colorTextMuted">
+                REVENUE SHARE RATIO
+              </Text>
+              <Text variant="h400" as="p">
+                Channel {token.revenueShareRatioPermill}%, Holders {100 - token.revenueShareRatioPermill}%
+              </Text>
+            </FlexBox>
+          </FlexBox>
+          <FlexBox flow="column">
+            <Text variant="h100" as="h1" color="colorTextMuted">
+              STAKED HOLDERS
+            </Text>
+            {activeRevenueShare?.stakers.length ? (
+              <StakersBox>
+                <FlexBox>
+                  {activeRevenueShare.stakers.slice(0, Math.min(activeRevenueShare.stakers.length, 5)).map((staker) => (
```

Comment:
generally, there won't be an error if you slice for more then and array length, simply the whole array will be returned, so I think the Math.min here is an overkill

Reply:
```suggestion
                  {activeRevenueShare.stakers.slice(0, 5).map((staker) => (
```